added gatsby image support

floydnoel opened the pull request:
This PR adds support for [gatsby-image](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-image#readme):

> Speedy, optimized images without the work.
> 
> gatsby-image is a React component specially designed to work seamlessly with Gatsby's GraphQL queries. It combines Gatsby's native image processing capabilities with advanced image loading techniques to easily and completely optimize image loading for your sites. gatsby-image uses gatsby-plugin-sharp to power its image transformations.

erquhart commented:
Thanks for looking into this!

The deploy preview for this branch shows lost and reduced quality images, can you look into it? Example:

Current: https://gatsby-netlify-cms.netlify.com/products

This PR: https://deploy-preview-117--gatsby-netlify-cms.netlify.com/products/

floydnoel commented:
Yes, I'll look into it and update the PR. Some of the image quality reduction is a result of the settings I chose; I'll bump up the quality parameters all around. Wasn't sure if there was a benefit to showing the compression available or not. Any other suggestions or feedback is welcome, thank you very much for taking the time to review my PR

floydnoel commented:
Fixed the image I missed initially, and adjusted the quality settings in the query. After checking between master and this branch I believe the image quality is roughly equivalent. There is a tiny bit less space between the two images and the large coffee bean image. Wasn't sure if the tradeoff of additional styling was worth the increased consistency, or not.

floydnoel commented on gatsby-config.js:
```diff
@@ -5,6 +5,14 @@ module.exports = {
+      resolve: 'gatsby-source-filesystem',
+      options: {
+        path: `${__dirname}/static/img`,
+        name: 'uploads',
```
this name should match the name on line 39. Not sure if `uploads` is best, maybe it should be `img` or `images`?

ZoltanVeres commented:
If you go to the Admin site it gives the following error: **Failed to load entries: TypeError: Cannot read property 'fluid' of undefined**. This kinda makes sence because the client side CMS and React doesn't know about the fluid property, i think this needs a workaround (see screenshots).

The only solution i could think of is to somehow replace that on the build step, just looking for `childImageShart.fluid` and the other combinations too and remove them from the template

But If somebody points me to a direction on how to fix this, i might be able to fix it....

<img width="826" alt="screen shot 2018-10-19 at 12 48 06" src="https://user-images.githubusercontent.com/214139/47211272-bc765d00-d39d-11e8-88fc-12b099c8605e.png">
<img width="546" alt="screen shot 2018-10-19 at 12 48 45" src="https://user-images.githubusercontent.com/214139/47211277-c009e400-d39d-11e8-8593-903a7c34ccd6.png">
<img width="654" alt="screen shot 2018-10-19 at 12 49 00" src="https://user-images.githubusercontent.com/214139/47211281-c304d480-d39d-11e8-9952-874d29317a65.png">

floydnoel commented:
Thanks @ZoltanVeres for pointing that out, I have a PR [here](https://github.com/floydnoel/gatsby-starter-netlify-cms/pull/2) which I'm validating and will submit here ASAP.

erquhart commented:
Sounds good 👍

floydnoel commented:
new PR here: 
https://github.com/netlify-templates/gatsby-starter-netlify-cms/pull/122
